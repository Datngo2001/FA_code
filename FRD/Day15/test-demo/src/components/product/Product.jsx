import React from "react";

function Product({ items }) {
  return (
    <div>
      <h2>Product</h2>
      {items?.length > 0 ? (
        items.map((item) => <p>{item.name}</p>)
      ) : (
        <div>No Item</div>
      )}
    </div>
  );
}

export default Product;
