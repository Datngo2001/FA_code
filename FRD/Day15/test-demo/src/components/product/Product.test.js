import renderer from "react-test-renderer"
import Product from "./Product"
describe("<Product/>", () => {
    it.only("should render correcttly when items is empty", () => {
        const tree = renderer.create(<Product />).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it("should render correcttly when items has value", () => {
        const tree = renderer.create(<Product items={["Honda", "BMW", "Volvo"]} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})