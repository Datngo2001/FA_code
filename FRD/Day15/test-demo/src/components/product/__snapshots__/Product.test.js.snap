// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product/> should render correcttly when items has value 1`] = `
<div>
  <h2>
    Product
  </h2>
  <p />
  <p />
  <p />
</div>
`;

exports[`<Product/> should render correcttly when items is empty 1`] = `
<div>
  <h2>
    Product
  </h2>
  <div>
    No Item
  </div>
</div>
`;
